### ISSUE TITLE
Improve `Breadcrumb` Accessibility

### ISSUE THREAD

**Issue by hahnwhitney** [vtex/faststore#1936]
**Background**
I've been working on a Jest/RTL accessibility test suite for FastStore's UI components. This issue isn't a bug in the sense that the component is "broken" -- it just doesn't meet the WCAG standards.

**Describe the bug -- Part 1**
It's obvious that you all have put a lot of work into making the `Breadcrumb` component accessible, so I think these things were likely just accidentally missed.

The [`<DropdownButton>` tag on line 141](https://github.com/vtex/faststore/blob/2847a365cb0da48e493ecdb0c9f2975e672628d4/packages/components/src/molecules/Breadcrumb/BreadcrumbBase.tsx#L141) of the `DropdownButton` component needs discernible text.

While "separators" (usually `...`, `>`, `/`, etc.) that are purely visual/decorative _do not_ need button text, the `<DropdownButton>` in the breadcrumb component actually serves a purpose: clicking it expands a dropdown menu; for this reason, we need to fix one of the following:
- Element does not have inner text that is visible to screen readers
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Element has no title attribute
- Element's default semantics were not overridden with role="none" or role="presentation"

**Possible Solution -- Part 1**
In this case, I think it makes sense to add an aria-label attribute (not sure about the specific wording though):
```
            <DropdownButton data-fs-breadcrumb-dropdown-button aria-label="Expand"size="small">
              {dropdownButtonIcon}
            </DropdownButton>
```

**Describe the bug -- Part 2**
The way the [breadcrumb navigation](https://github.com/vtex/faststore/blob/2847a365cb0da48e493ecdb0c9f2975e672628d4/packages/components/src/molecules/Breadcrumb/Breadcrumb.tsx#L7) is written, we're duplicating a lot of code:
```
const Breadcrumb = ({ breadcrumbList, ...otherProps }: BreadcrumbProps) => (
  <>
    <BreadcrumbBase breadcrumbList={breadcrumbList} {...otherProps} />
    <BreadcrumbBase breadcrumbList={breadcrumbList} isDesktop {...otherProps} />
  </>
)
```

Line 9 (`<BreadcrumbBase breadcrumbList={breadcrumbList} {...otherProps} />` generates the _not-desktop_ breadcrumb nav; line 10 (`<BreadcrumbBase breadcrumbList={breadcrumbList} isDesktop {...otherProps} />`) generates the _desktop_ breadcrumb -- but the DOM output from these two lines is almost exactly the same, with one additional piece included in the desktop nav:
![Screenshot 2023-08-07 at 12 29 00 PM](https://github.com/vtex/faststore/assets/88338049/1c9b4a1d-d205-4727-826e-61a6acdeeaba)

**Possible Solution -- Part 2**
The accessibility issue here is that the two versions of the breadcrumb nav are not unique (meaning, there is no way to distinguish the duplicate items in the mobile vs. desktop nav items since they both appear in the DOM at the same time.

I think a better way to fix this might be to avoid rendering two copies of the breadcrumbs in the first place. We could still leave the `data-fs-breadcrumb-is-desktop` boolean, but maybe we could add a check that would only render desktop elements if the `isDesktop` boolean is true?

**Comment by renatamottam** [comment]
Hi @hahnwhitney, thanks for reporting this issue in such detail, it's really helpful ✨
We created an internal task and will fix it soon.

**Comment by renatamottam** [comment]
Updates about the issue:
- We've fixed the `aria-label` bug and it's already available on fs core version `2.1.81`. ✨ 
- Although it's a relevant improvement to only show one breadcrumb component on DOM, it's not an accessibility issue as we're using the [visibility: hidden](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#accessibility_concerns) property, that way the element and all its descendant elements are no longer announced by screen reading technology. We created a task to improve this component's rendering into the DOM and will come back to this on the future.
